### PR TITLE
Windows UWP: Add support for emulated mouse cursor for targets such as XBOX

### PR DESCRIPTION
**Pull request by clarkezone**
Adds an emulated mouse cursor for devices that don't typically have a traditional mouse.  Enables mouse-like user interactions by using gamepad devices. 

https://github.com/flutter/flutter/issues/76092
https://github.com/flutter/flutter/issues/14967

Unit tests blocked by: https://github.com/flutter/flutter/issues/70197 label added

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [ ] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on
writing and running engine tests.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.
- [x] The reviewer has submitted any presubmit flakes in this PR using the [engine presubmit flakes form] before re-triggering the failure.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/master/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[engine presubmit flakes form]: https://forms.gle/Wc1VyFRYJjQTH6w5A

### CODE REVIEW

File: shell/platform/windows/game_pad_winuwp.cc
Comment (cbracken) [inline shell/platform/windows/game_pad_winuwp.cc:98]: nit: just `return value > 0.1 || value < -0.1;`

File: shell/platform/windows/game_pad_winuwp.cc
Comment (cbracken) [inline shell/platform/windows/game_pad_winuwp.cc:112]: This appears to be unused.

File: shell/platform/windows/game_pad_winuwp.cc
Comment (cbracken) [inline shell/platform/windows/game_pad_winuwp.cc:129]: Here and for the right stick: this looks like `RaiseLeftStickMoved` will be called so long as the values are between -0.1 and 0.1, even if they haven't changed. Is that expected?

The trigger code looks more like what I'd expect.

File: shell/platform/windows/game_pad_winuwp.cc
Comment (cbracken) [inline shell/platform/windows/game_pad_winuwp.cc:175]: Here and below: consider merging the outer two conditionals to reduce nesting.

File: shell/platform/windows/game_pad_winuwp.cc
Comment (cbracken) [inline shell/platform/windows/game_pad_winuwp.cc:84]: Could this method be `const`?

File: shell/platform/windows/game_pad_winuwp.cc
Comment (cbracken) [inline shell/platform/windows/game_pad_winuwp.cc:47]: Could this be `const`?

You'd need to mark the mutex as `mutable` but that's encouraged. IIRC the "M&M rule" -- mutable and mutex go together :)

**Comment by chinmaygarde** [comment]
@clarkezone Any updates on @cbracken review comments?

**Comment by clarkezone** [comment]
Appols, I was waiting for some of the other UWP commits to land which they now have, was looking at the template in the meantime.  I'll take a look this weekend, likely tomorrow.

File: shell/platform/windows/game_pad_cursor_winuwp.h
Comment (clarkezone) [reply shell/platform/windows/game_pad_cursor_winuwp.h:36]: fixed in next commit

File: shell/platform/windows/game_pad_cursor_winuwp.h
Comment (clarkezone) [reply shell/platform/windows/game_pad_cursor_winuwp.h:37]: Fixed in next commit.

File: shell/platform/windows/game_pad_cursor_winuwp.h
Comment (clarkezone) [reply shell/platform/windows/game_pad_cursor_winuwp.h:46]: Fixed in next commit.

File: shell/platform/windows/game_pad_cursor_winuwp.cc
Comment (clarkezone) [reply shell/platform/windows/game_pad_cursor_winuwp.cc:113]: Fixed in next commit.

File: shell/platform/windows/game_pad_winuwp.cc
Comment (clarkezone) [reply shell/platform/windows/game_pad_winuwp.cc:94]: Fixed in next commit

File: shell/platform/windows/game_pad_winuwp.cc
Comment (clarkezone) [reply shell/platform/windows/game_pad_winuwp.cc:112]: Fixed in next commit.

**Comment by chinmaygarde** [comment]
@cbracken Your comments seems to have been addressed (not sure if completely so). Can you re-review please so we can unblock?

**Comment by clarkezone** [comment]
I haven't been through them all yet, I plan on finishing up the review by the end of the weekend.

File: shell/platform/windows/display_helper_winuwp.h
Comment (clarkezone) [reply shell/platform/windows/display_helper_winuwp.h:34]: Good feedback, fixed in next commit.

File: shell/platform/windows/game_pad_winuwp.cc
Comment (clarkezone) [reply shell/platform/windows/game_pad_winuwp.cc:84]: Yes, it's a private method.

File: shell/platform/windows/game_pad_cursor_winuwp.cc
Comment (clarkezone) [reply shell/platform/windows/game_pad_cursor_winuwp.cc:97]: Fixed in next commit.

File: shell/platform/windows/game_pad_cursor_winuwp.h
Comment (clarkezone) [reply shell/platform/windows/game_pad_cursor_winuwp.h:53]: Ultimately, we will want more control eg turn mouse emulation on and off, off for the case where the flutter app is consuming the gamecontroller input directly rather than us emulating a mouse and injecting mouse events.  ATM this change is an MVP to stand up basic xbox support, would prefer to not grow the scope in this PR particularly as that would require framework integration.

File: shell/platform/windows/game_pad_cursor_winuwp.h
Comment (clarkezone) [reply shell/platform/windows/game_pad_cursor_winuwp.h:56]: Fixed in next commit.

File: shell/platform/windows/game_pad_cursor_winuwp.h
Comment (clarkezone) [reply shell/platform/windows/game_pad_cursor_winuwp.h:79]: man you have eagle eyes ;-)  Fixed in next commit.

File: shell/platform/windows/game_pad_cursor_winuwp.h
Comment (clarkezone) [reply shell/platform/windows/game_pad_cursor_winuwp.h:83]: Fixed in next commit.

File: shell/platform/windows/game_pad_cursor_winuwp.h
Comment (clarkezone) [reply shell/platform/windows/game_pad_cursor_winuwp.h:101]: Fixed in next commit.

File: shell/platform/windows/game_pad_winuwp.cc
Comment (clarkezone) [reply shell/platform/windows/game_pad_winuwp.cc:127]: Yes because you are holding the stick at a constant angle, hence the value is constant, but you want the cursor to keep moving so you need to keep sending an update at polling frequency for it to keep moving.  Make sense?

File: shell/platform/windows/display_helper_winuwp.h
Comment (clarkezone) [reply shell/platform/windows/display_helper_winuwp.h:52]: Fixed in next commit.

File: shell/platform/windows/display_helper_winuwp.h
Comment (clarkezone) [reply shell/platform/windows/display_helper_winuwp.h]: Fixed in next commit.

File: shell/platform/windows/display_helper_winuwp.h
Comment (clarkezone) [reply shell/platform/windows/display_helper_winuwp.h]: Fixed in next commit

File: shell/platform/windows/flutter_window_winuwp.cc
Comment (clarkezone) [reply shell/platform/windows/flutter_window_winuwp.cc]: I've cleaned this up a bit.  Answering why inverse transform not required for XBOX would require a writeup on how DPI works in UWP apps as a baseline to be able to then explain the delta.  I can do this but would like to get through the rest of this PR first.

File: shell/platform/windows/game_pad_cursor_winuwp.cc
Comment (clarkezone) [reply shell/platform/windows/game_pad_cursor_winuwp.cc:147]: As per the inverse transform, the scaling has to be conditional for the case when running on xbox aka big screen devices.  However, agree that having lots of the same conditional scattered through the code is not nice hence moving into a helper to centralize this logic in next commit.

File: shell/platform/windows/game_pad_cursor_winuwp.h
Comment (clarkezone) [reply shell/platform/windows/game_pad_cursor_winuwp.h:102]: Fixed in next commit.

File: shell/platform/windows/game_pad_cursor_winuwp.cc
Comment (clarkezone) [reply shell/platform/windows/game_pad_cursor_winuwp.cc:282]: Fixed in next commit.

File: shell/platform/windows/game_pad_winuwp.cc
Comment (clarkezone) [reply shell/platform/windows/game_pad_winuwp.cc:98]: Fixed in next commit

File: shell/platform/windows/game_pad_winuwp.cc
Comment (clarkezone) [reply shell/platform/windows/game_pad_winuwp.cc:173]: Fixed in next commit.

File: shell/platform/windows/game_pad_winuwp.cc
Comment (clarkezone) [reply shell/platform/windows/game_pad_winuwp.cc:84]: Fixed in next commit

File: shell/platform/windows/game_pad_winuwp.cc
Comment (clarkezone) [reply shell/platform/windows/game_pad_winuwp.cc:47]: Fixed in next commit.

File: shell/platform/windows/game_pad_cursor_winuwp.h
Comment (clarkezone) [reply shell/platform/windows/game_pad_cursor_winuwp.h]: Fixed in next commit.

File: shell/platform/windows/game_pad_cursor_winuwp.h
Comment (clarkezone) [reply shell/platform/windows/game_pad_cursor_winuwp.h]: Fixed in next commit.

File: shell/platform/windows/display_helper_winuwp.h
Comment (clarkezone) [reply shell/platform/windows/display_helper_winuwp.h:24]: Doing this will require 4 casts from double to float in different consuming sites.  Happy to do that if you would like.

File: shell/platform/windows/display_helper_winuwp.cc
Comment (clarkezone) [reply shell/platform/windows/display_helper_winuwp.cc:76]: Fixed in the next commit.

**Comment by clarkezone** [comment]
OK I'm done with changes..

File: shell/platform/windows/display_helper_winuwp.h
Comment (cbracken) [reply shell/platform/windows/display_helper_winuwp.h:24]: Ah good point. I'm going to need to make that change separately for a11y support. At that point, we can consolidate across all the bits of the code that are implementing something along these lines. This lgtm as-is then!

**Review by cbracken — APPROVED** [review]
lgtm on pass two! Thanks for the fixes!

**Comment by cbracken** [comment]
Rebased against tip-of-tree after the software-rendering fallback landed.

**Pull request merged**
